Foundation/FileManager+Win32.swift

```diff
@@ -699,18 +699,16 @@ extension FileManager.NSPathDirectoryEnumerator {
                 (fsr.flatMap { String(utf8String: $0) })?.withCString(encodedAs: UTF16.self) { f($0) }
             }
         }
-        let result = withURLCString(url: baseURL) { pszFrom -> BOOL? in
+        guard let _ = withURLCString(url: baseURL, { pszFrom -> Bool? in
             withURLCString(url: url) { pszTo in
                 let fromAttrs = GetFileAttributesW(pszFrom)
                 let toAttrs = GetFileAttributesW(pszTo)
                 guard fromAttrs != INVALID_FILE_ATTRIBUTES, toAttrs != INVALID_FILE_ATTRIBUTES else {
-                    return FALSE
+                    return false
                 }
                 return PathRelativePathToW(relativePath.baseAddress, pszFrom, fromAttrs, pszTo, toAttrs)
             }
-        }
-
-        guard result == TRUE, let (path, _) = String.decodeCString(relativePath.baseAddress, as: UTF16.self) else {
```

Comment:
I think you still have to bind the withURLCString to a result and check the result. Else it'll check that the you got non nil, but won't check the actual value, I don't believe it gets implicitly flattened. Perhaps
```
guard let result = ...
     }), result, let(path, _) ...
```